lib/processor.js

```diff
@@ -370,14 +373,28 @@ function excludeUnsatisfiableRules(message) {
  * Transforms generated messages for output.
  * @param {Array<Message[]>} messages An array containing one array of messages
  *     for each code block returned from `preprocess`.
+ * @param {string} filename The filename of the file
  * @returns {Message[]} A flattened array of messages with mapped locations.
  */
-function postprocess(messages) {
-    return [].concat(...messages.map((group, i) => {
-        const adjust = adjustBlock(blocks[i]);
+function postprocess(messages, filename) {
+    const blocks = blocksCache.get(filename) || [];
+
+    const filteredMessages = [].concat(...messages.map((group, i) => {
+        const block = blocks[i];
+
+        // non code block message, parsed by `eslint-mdx` for example
+        if (!block) {
```

Comment:
Can you elaborate on what scenarios could cause a cache miss above (requiring the `|| []` fallback) or `block` to be missing here? At least so far in this processor's history, those would both indicate bugs where there's a mismatch between the logic in `preprocess()` and `postprocess()`. If that's still the case, I'd like to fail as close to the source as possible so that we get the most accurate stack traces and bug reports. While the fallbacks are nice in that they probably prevent a crash, they could lead to undefined behavior that shows incorrect results.

Reply:
It's about integration with `eslint-plugin-mdx`, not a bug for `eslint-plugin-markdown`, I want to reuse `eslint-plugin-markdown`'s processor in `eslint-plugin-mdx`.

See https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/src/processors/remark.ts#L29

Reply:
Ah, I see that `eslint-plugin-mdx` appends one additional virtual file from `preprocess()`. In `eslint-plugin-mdx`'s `postprocess()`, could you then trim the last `Message[]` from `lintMessages`? I think that would ensure that `eslint-plugin-markdown`'s `postprocess()` here would only get messages that came from its own `preprocess()`, and these fallbacks wouldn't be needed anymore.

As-is, these fallbacks are implementation details around error handling in the `postprocess()` method. They're not part of the public API mandated by ESLint. A refactor of this logic could hypothetically change this behavior in a way that's fully compatible with the public API yet unintentionally invalidates an assumption that `eslint-plugin-mdx` was making. That's why I'd prefer to have the "extra block" behavior fully owned and controlled within `eslint-plugin-mdx` if at all possible.

Reply:
Thanks for your great suggestion! I'll refactor it ASAP.

Reply:
@btmills Done!